### PR TITLE
Upgrade to libuv trunk

### PR DESCRIPTION
**Pull request by tlbdk**
Hi,

I did some quick fixes to make it compile against the newest libuv.

Regards Troels

### CODE REVIEW

File: webserver.c [hunk 1]
```diff
@@ -43,33 +42,30 @@ void on_close(uv_handle_t* handle) {
...
-    parsed = http_parser_execute(
-        &client->parser, &parser_settings, buf.base, nread);
+    parsed = http_parser_execute(&client->parser, &parser_settings, buf->base, nread);
     if (parsed < nread) {
       LOG_ERROR("parse error");
       uv_close((uv_handle_t*) &client->handle, on_close);
     }
   } else {
-    uv_err_t err = uv_last_error(uv_loop);
+    /* uv_err_t err = uv_last_error(uv_loop);
```
Comment (philips) [inline webserver.c:62]: It would probably be better to just delete this I suppose.

Comment (philips) [inline webserver.c:62]: How are errors checked now in the new version. This is just empty now?

**Comment by philips** [comment]
I appreciate the PR but I will have to figure out error handling :)

**Comment by tlbdk** [comment]
I think the the new idea is that all functions return a error code and then
you can translate that in to a string. But how that works with the
callbacks I have clue. I'm just starting with libuv and I was looking for a
good working sample to get started, your code was a good starting point for
what I want to do. I'm working on a small http(s) proxy that supports rfc
2817 - http SSL upgrade, for our airprint gateway product. Hopefully I will
get something working that also does error handling so I can also share
that too, but if you beat me to it I won't complain :)

But thanks for the code as it's hard to find working samples that are also
easy to understand.

Regards Troels

**Comment by tlbdk** [comment]
I seems this will not get merged.

**Pull request closed without merging**